File: bbarena-model/src/main/java/net/sf/bbarena/model/event/Die.java
package net.sf.bbarena.model.event;

import java.io.Serializable;

public enum Die implements Serializable {

	D2(2),
	D3(3),
	D4(4),
	D6(6),
	D8(8),
	D10(10),
	D11(11),
	D12(12),
	D16(16),
	D100(100),
	DB(6), // Block Die
	DS(8), // Scatter Die
	D68(6, 8); // Special CAS Die
	
	private static final long serialVersionUID = -1269858379098579914L;

	private Integer[] _faces;

	Die(Integer... faces) {
		_faces = faces;
	}

	public Integer[] getFaces() {
		return _faces;
	}

}

File: src/main/kotlin/bbarena/model/event/Die.kt
package bbarena.model.event

import java.io.Serializable

enum class Die(val faces: Array<Int>) : Serializable {

    D2(arrayOf(2)),
    D3(arrayOf(3)),
    D4(arrayOf(4)),
    D6(arrayOf(6)),
    D8(arrayOf(8)),
    D10(arrayOf(10)),
    D11(arrayOf(11)),
    D12(arrayOf(12)),
    D16(arrayOf(16)),
    D100(arrayOf(100)),
    DB(arrayOf(6)), // Block Die
    DS(arrayOf(8)), // Scatter Die
    D68(arrayOf(6, 8));

}
